gamemodes/terrortown/gamemode/server/sv_player_ext.lua

```diff
@@ -787,119 +789,191 @@ local function FindCorpse(ply)
-
-			onFail(ply)
+		else
+			OnReviveFailed(ply, "message_revival_failed")
 		end
 	end)
 end
 
 ---
--- Returns if a player is currently in a revival process started by @{Player:Revive}
--- @return boolean The revival status
+-- Cancel the ongoing revival process.
+-- @param [default="message_revival_canceled"]string failMessage The fail message that should be displayed for the client
+-- @realm server
+function plymeta:CancelRevival(failMessage)
+	self:SetReviving(false)
+	self:SetBlockingRevival(false)
+	self:SetRevivalReason(nil)
+
+	timer.Remove("TTT2RevivePlayer" .. self:EntIndex())
+
+	OnReviveFailed(self, failMessage or "message_revival_canceled")
+end
+
+---
+-- Sets the revival state.
+-- @param [default=false]boolean isReviving The reviving state
+-- @realm server
+function plymeta:SetReviving(isReviving)
+	self:TTT2NETSetBool("player_is_reviving", isReviving or false)
+end
+
+---
+-- Sets the blocking revival state
+-- @param [default=false]boolean isBlockingRevival The blocking revival state
+-- @realm server
+function plymeta:SetBlockingRevival(isBlockingRevival)
+	self:TTT2NETSetBool("player_is_blocking_revival", isBlockingRevival or false)
+end
+
+---
+-- Sets the revival start time
+-- @param [default=@{CurTime()}]number startTime The revival start time
+-- @realm server
+function plymeta:SetRevivalStartTime(startTime)
+	self:TTT2NETSetFloat("player_revival_start_time", startTime or CurTime())
+end
+
+---
+-- Sets the revival time
+-- @param [default=0.0]number time The revival time
+-- @realm server
+function plymeta:SetRevivalTime(time)
+	self:TTT2NETSetFloat("player_revival_time", time or 0.0)
+end
+
+---
+-- Sets the death position
+-- @param Vector pos The death position
 -- @realm server
-function plymeta:IsReviving()
-	return self.reviving or false
+function plymeta:SetDeathPosition(pos)
+	self:TTT2NETSetFloat("player_death_pos_x", pos.x or 0.0)
+	self:TTT2NETSetFloat("player_death_pos_y", pos.y or 0.0)
+	self:TTT2NETSetFloat("player_death_pos_z", pos.z or 0.0)
+end
+
+---
+-- Sets the spawn position
+-- @param Vector pos The spawn position
+-- @realm server
+function plymeta:SetSpawnPosition(pos)
+	self:TTT2NETSetFloat("player_spawn_pos_x", pos.x or 0.0)
+	self:TTT2NETSetFloat("player_spawn_pos_y", pos.y or 0.0)
+	self:TTT2NETSetFloat("player_spawn_pos_z", pos.z or 0.0)
+end
+
+---
+-- Sets the if a player was active (TEAM_TERROR) in a round.
+-- @param boolean state The state
+-- @realm server
+function plymeta:SetActiveInRound(state)
+	if state and (GetRoundState() ~= ROUND_ACTIVE or not self:IsTerror()) then return end
+
+	self:TTT2NETSetBool("player_was_active_in_round", state or false)
+end
+
+---
+-- Sets the if a player died in a round.
+-- @param boolean state The state
+-- @realm server
+function plymeta:SetDiedInRound(state)
+	if state and (GetRoundState() ~= ROUND_ACTIVE or not self:IsTerror()) then return end
+
+	self:TTT2NETSetBool("player_has_died_in_round", state or false)
```

Comment:
Please note, that this information is broadcasted to all players

Reply:
yes. This info is needed on all clients for the respawn indicator